### PR TITLE
Fixes issue gh-153 - multiple bootstrap contexts with hierarchy

### LINKED ISSUE #153: Seeing multiple bootstrap contexts with Spring Boot hierarchy
There is likely a gap in my understanding here. When I create a spring cloud based application with a hierarchy, say the following:

```java
SpringApplicationBuilder appBuilder =
                new SpringApplicationBuilder()
                        .parent(RootContext.class)
                        .child(ChildContext.class);

        ConfigurableApplicationContext applicationContext  = appBuilder.run();
```

I am seeing **two bootstrap** contexts, visualized here using the `/beans` actuator endpoint:

screen shot 2016-12-17 at 1 55 19 pm

My expectation was to see exactly one bootstrap context.

### PR DESCRIPTION
**Pull request by bijukunjummen**
Introduced additional checks to ensure that the bootstrap context is not already present in the hierarchy before adding one. Attempts to fix #153

### CODE REVIEW

File: spring-cloud-context/src/test/java/org/springframework/cloud/bootstrap/BootstrapListenerHierarchyIntegrationTests.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Integration tests for Bootstrap Listener
+ * 
+ * @author Biju Kunjummen
+ */
+public class BootstrapListenerHierarchyIntegrationTests {
+
+	private ConfigurableApplicationContext context;
```
Comment (spencergibb) [inline spring-cloud-context/src/test/java/org/springframework/cloud/bootstrap/BootstrapListenerHierarchyIntegrationTests.java]: No need for a field here.

Comment (bijukunjummen) [inline spring-cloud-context/src/test/java/org/springframework/cloud/bootstrap/BootstrapListenerHierarchyIntegrationTests.java]: Thanks @spencergibb, this is fixed -  the variable is now local to each test.

**Comment by spencergibb** [comment]
ping @dsyer

**Comment by spencergibb** [comment]
@bijukunjummen will you rebase?

**Comment by bijukunjummen** [comment]
Done @spencergibb.

**Comment by dsyer** [comment]
Fixed (I think in a better way, but only time will tell) in 331e790. I borrowed the test case though, thanks!

**Comment by bijukunjummen** [comment]
Awesome, yes your fix appears way simpler @dsyer

**Pull request closed without merging**